cli/cfncluster/cfncluster.py

```diff
@@ -346,6 +371,51 @@ def instances(args):
     for instance in instances:
         print('%s         %s' % (instance[0],instance[1]))
 
+def get_head_user(parameters, template):
+    mappings = template.get("TemplateBody") \
+            .get("Mappings") \
+            .get("OSFeatures")
+    base_os =[i.get('ParameterValue') for i in parameters if i.get('ParameterKey') == "BaseOS"][0]
+    return mappings.get(base_os).get("User")
+
+def command(args, extra_args):
+    stack = ('cfncluster-' + args.cluster_name)
+    config = cfnconfig.CfnClusterConfig(args)
+    if args.command in config.aliases:
+        config_command = config.aliases[args.command]
+    else:
+        config_command = "ssh {CFN_USER}@{MASTER_IP} {ARGS}"
+
+    cfn = boto3.client('cloudformation', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
+    try:
+        status = cfn.describe_stacks(StackName=stack).get("Stacks")[0].get('StackStatus')
+        invalid_status = ['DELETE_COMPLETE', 'DELETE_IN_PROGRESS']
+        if status in invalid_status:
+            logger.info("Stack status: %s. Cannot SSH while in %s" % (status, ' or '.join(invalid_status)))
+            sys.exit(1)
+        ip = get_master_server_ip(stack, config)
+        stack_result = cfn.describe_stacks(StackName=stack).get('Stacks')[0]
+        template = cfn.get_template(StackName=stack)
+        username = get_head_user(stack_result.get('Parameters'), template)
+
+        # build command
+        cmd = config_command.format(CFN_USER=username, MASTER_IP=ip, ARGS=' '.join(str(e) for e in extra_args))
```

Comment:
I think `extra_args` with spaces, e.g. `-i "/path with spaces/to/pemfile"` is not going to work.
I think the python3 function `shlex.quote()` on the `str(e)` does the job.

Reply:
fixed. `shlex.quote()` was introduced in 3.3 so I've added code to import either `pipes.quote` or `shlex.quote`